pay-api/src/pay_api/services/eft_short_name_summaries.py

```diff
@@ -0,0 +1,136 @@
+# Copyright © 2024 Province of British Columbia
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Service to provide summarized information for EFT Short names."""
+from __future__ import annotations
+
+from flask import current_app
+from sqlalchemy import and_, func, or_
+
+from pay_api.models import EFTCredit as EFTCreditModel
+from pay_api.models import EFTShortnameLinks as EFTShortnameLinksModel
+from pay_api.models import EFTShortnames as EFTShortnameModel
+from pay_api.models import EFTShortnameSummarySchema as EFTSummarySchema
+from pay_api.models import EFTTransaction as EFTTransactionModel
+from pay_api.models import db
+from pay_api.services.eft_short_names import EFTShortnamesSearch
+from pay_api.utils.converter import Converter
+from pay_api.utils.enums import EFTFileLineType, EFTProcessStatus, EFTShortnameStatus
+
+
+class EFTShortnameSummaries:
+    """Service to provide summarized information for EFT Short names."""
+
+    @classmethod
+    def search(cls, search_criteria: EFTShortnamesSearch):
+        """Search eft short name summaries."""
+        current_app.logger.debug('<search')
+        search_count = cls.get_search_count()
+        search_query = cls.get_search_query(search_criteria)
+        pagination = search_query.paginate(per_page=search_criteria.limit,
+                                           page=search_criteria.page)
+
+        summary_list = [EFTSummarySchema.from_row(short_name_summary) for short_name_summary in pagination.items]
```

Comment:
I've seen this code repeated in a few services, might be up for helper function?

Reply:
Added an unstructure helper.

I have some work on another EFT search route, I will do some clean up there and update anywhere that can use this helper.